app/transport/http/user.go

```diff
@@ -35,20 +36,23 @@ func (uh UserHandler) CreateUser() http.Handler {
 		if err := user.Validate(); err != nil {
 			WriteJSONResponse(w, http.StatusBadRequest, Response{Message: BadRequestMsg, Details: err.Error()}, uh.logger)
 			uh.logger.Errorf("Failed validating user: %+v", err)
+
 			return
 		}
 
 		id, err := uh.usecase.Signup(user)
 		if err != nil {
 			uh.logger.Errorf("Failed creating user: %+v", err)
 
-			// TODO: Handle errors gracefully.
-			if err, ok := errors.Unwrap(err).(*pq.Error); ok && err.Code.Name() == "unique_violation" {
+			// Handle errors gracefully.
```

Comment:
If you already handle the errors gracefully - maybe we should remove this comment.